litestar/utils/typing.py

```diff
@@ -249,7 +249,9 @@ def get_type_hints_with_generics_resolved(
     if origin is None:
         # Implies the generic types have not been specified in the annotation
         type_hints = get_type_hints(annotation, globalns=globalns, localns=localns, include_extras=include_extras)
-        typevar_map = {p: p for p in annotation.__parameters__}
+        if not (parameters := getattr(annotation, "__parameters__", None)):
+            return type_hints
+        typevar_map = {p: p for p in parameters}
```

Comment:
This is to support using this function without knowing up front if the type we are passing in is a generic type or not. If not, it won't have the parameters attribute, and so we just return the type hints without any post processing.